src/test/scala/com/microsoft/hyperspace/index/IndexManagerTests.scala

```diff
@@ -264,10 +267,24 @@ class IndexManagerTests extends HyperspaceSuite with SQLHelper {
           .parquet(testPath)
         val df = spark.read.parquet(testPath)
         hyperspace.createIndex(df, indexConfig)
-        val ex = intercept[HyperspaceException] {
-          hyperspace.refreshIndex(indexConfig.indexName)
+        val indexPath = PathUtils.makeAbsolute(s"$systemPath/${indexConfig.indexName}")
+        val logManager = IndexLogManagerFactoryImpl.create(indexPath)
+        val latestId = logManager.getLatestId().get
+
+        MockEventLogger.reset()
+        hyperspace.refreshIndex(indexConfig.indexName)
+        // Check that no new log files were created in this operation.
+        assert(latestId == logManager.getLatestId().get)
+
+        // Check emitted events.
+        MockEventLogger.emittedEvents match {
+          case Seq(
+              RefreshAppendActionEvent(_, _, "Operation started."),
+              RefreshAppendActionEvent(_, _, msg))
+              if msg.equals("No-op operation recorded: Refresh append aborted as" +
+                " no appended source data files found.") => // pass
```

Comment:
How about the following (preferring explicit assert):
```suggestion
          case Seq(
              RefreshDeleteActionEvent(_, _, "Operation started."),
              RefreshDeleteActionEvent(_, _, msg)) =>
            assert(msg.contains("Refresh delete aborted as no deleted source data file found"))
```